kernel/src/time/clocks/system_wide.rs

```diff
@@ -144,47 +144,75 @@ impl BootTimeClock {
 }
 
 impl Clock for JiffiesClock {
-    fn read_time(&self) -> Duration {
-        Jiffies::elapsed().as_duration()
+    fn read_duration(&self) -> Duration {
+        self.read_jiffies().into()
+    }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::elapsed()
     }
 }
 
 impl Clock for RealTimeClock {
-    fn read_time(&self) -> Duration {
+    fn read_duration(&self) -> Duration {
         SystemTime::now()
             .duration_since(&SystemTime::UNIX_EPOCH)
             .unwrap()
     }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::from_duration(self.read_duration())
+    }
 }
 
 impl Clock for MonotonicClock {
-    fn read_time(&self) -> Duration {
+    fn read_duration(&self) -> Duration {
         read_monotonic_time()
     }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::from_duration(self.read_duration())
+    }
 }
 
 impl Clock for RealTimeCoarseClock {
-    fn read_time(&self) -> Duration {
+    fn read_duration(&self) -> Duration {
         *Self::current_ref().get().unwrap().disable_irq().lock()
     }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::from_duration(self.read_duration())
+    }
 }
 
 impl Clock for MonotonicCoarseClock {
-    fn read_time(&self) -> Duration {
-        RealTimeCoarseClock::get().read_time() - *START_TIME_AS_DURATION.get().unwrap()
+    fn read_duration(&self) -> Duration {
+        RealTimeCoarseClock::get().read_duration() - *START_TIME_AS_DURATION.get().unwrap()
+    }
+
+    fn read_jiffies(&self) -> Jiffies {
+        Jiffies::from_duration(self.read_duration())
     }
 }
 
 impl Clock for MonotonicRawClock {
-    fn read_time(&self) -> Duration {
+    fn read_duration(&self) -> Duration {
         read_monotonic_time()
     }
+
+    fn read_jiffies(&self) -> Jiffies {
```

Comment:
Due to my previous comment:
> If `Jiffies` is more like an [`Instant`](https://doc.rust-lang.org/std/time/struct.Instant.html) (which also has an [`elapsed`](https://doc.rust-lang.org/std/time/struct.Instant.html#method.elapsed) method), perhaps it shouldn't implement `Add`.

I tend to avoid the use of `Jiffies` here. Maybe we can define a new type, `JiffiesCount`, otherwise, we can just use `u64`.